server/channels/api4/post.go

```diff
@@ -1083,11 +1083,11 @@ func unpinPost(c *Context, w http.ResponseWriter, _ *http.Request) {
 
 func acknowledgePost(c *Context, w http.ResponseWriter, r *http.Request) {
 	// license check
-	permissionErr := minimumProfessionalLicense(c)
-	if permissionErr != nil {
-		c.Err = permissionErr
+	if !model.MinimumProfessionalLicense(c.App.Srv().License()) {
+		c.Err = model.NewAppError("", model.NoTranslation, nil, "license is neither professional nor enterprise nor premium", http.StatusNotImplemented)
```

Comment:
maybe something like this would be more future proof. Also, how does cloud license work? Is it enterprise?
```suggestion
		c.Err = model.NewAppError("", model.NoTranslation, nil, "feature is not available for the current license", http.StatusNotImplemented)
```

Reply:
I believe cloud is enterprise but not sure. I've updated the text.

Reply:
@nickmisasi do you know, is Cloud considered as enterprise? Also, is Premium will be available only on self-hosted?

Reply:
@nickmisasi or @marianunez can confirm this but from what I understand, if we wanna differentiate a behavior in cloud vs on-prem, we use the `license.IsCloud()` method. For restricting features based on SKUs, we just use the short SKU name checks. This is what I see commonly in code for feature restrictions.

Reply:
Correct, the SKUs are still the same for Cloud (could be Pro, Enterprise or this new one). There is a specific `isCloud` flag in cloud licenses that is what is used for any cloud specific checks.

Reply:
+1, thanks Maria. Sorry, didn't get notified for this.